### PR TITLE
Basically done, missing cascade on all objects

### PR DESCRIPTION
**Pull request by jubiiz**

### CODE REVIEW

File: backend/cranki/src/main/java/ca/mcgill/cranki/model/LiteralProperty.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package ca.mcgill.cranki.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.OneToOne;
+
+@Entity
+public class LiteralProperty extends Property{
+  @OneToOne(mappedBy = "property")
+  private PropertyValue value;
```
Comment (jubiiz) [inline backend/cranki/src/main/java/ca/mcgill/cranki/model/LiteralProperty.java:9]: does a literal property need a value? Should this value be tied to a `SpecificProperty`?

**Review by jubiiz — COMMENTED** [review]
~There seems to be no class for "SelectProperty". There is multiselect, but where is Select?~

EDIT: Multiselect property behaves the same as select property. Different behavior can occur in specific properties, e.g. only being allowed to select one. Could add a "max_num_select" attribute to multiselect properties to handle this case.

File: backend/cranki/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 spring.application.name=cranki
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-spring.jpa.hibernate.ddl-auto=update
+spring.jpa.hibernate.ddl-auto=create
```
Comment (wangwiza) [inline backend/cranki/src/main/resources/application.properties:3]: Have you discussed with the team about potential issues with this? How are people gonna get initial values in their database without manually entering them again?

File: backend/cranki/src/test/java/ca/mcgill/cranki/unitTest/PropertyStuffsTests.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  @Autowired
+  PropertyValueRepository propertyValueRepository;
+
+  @Autowired
+  SpecificPropertyRepository specificPropertyRepository;
+
+
+  @Test
+  public void propertiesStuffWorks() {
```
Comment (wangwiza) [inline backend/cranki/src/test/java/ca/mcgill/cranki/unitTest/PropertyStuffsTests.java:40]: Test name is ambiguous

**Review by wangwiza — APPROVED** [review]
??? In @jubiiz we trust?

File: backend/cranki/src/main/resources/application.properties (hunk 2 above)
Comment (vladarama) [reply backend/cranki/src/main/resources/application.properties:3]: @wangwiza 
Agreed, I think we should keep update. Or if we change it to create, we should write a data loader function which would load all the values in the DB at launch.

File: backend/cranki/src/main/resources/application-test.properties [hunk 4]
```diff
@@ -9,5 +9,6 @@ spring.datasource.url=jdbc:mysql://localhost:3306/test?createDatabaseIfNotExist=
 spring.datasource.username=root
 spring.datasource.password=lala
 spring.jpa.show-sql=true
+spring.jpa.properties.hibernate.enable_lazy_load_no_trans=true
```
[the remark this answers is not in the document]
Comment (jubiiz) [reply backend/cranki/src/main/resources/application-test.properties:12]: @timmyhoa ?

File: backend/cranki/src/main/resources/application.properties (hunk 2 above)
Comment (jubiiz) [reply backend/cranki/src/main/resources/application.properties:3]: @timmyhoa ?

File: backend/cranki/src/test/java/ca/mcgill/cranki/unitTest/PropertyStuffsTests.java (hunk 3 above)
Comment (jubiiz) [reply backend/cranki/src/test/java/ca/mcgill/cranki/unitTest/PropertyStuffsTests.java:40]: @timmyhoa ?

**Pull request merged**